app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java

```diff
@@ -0,0 +1,576 @@
+                application.setOrganizationId(null);
+                examplesOrganizationCloner.makePristine(application);
+                String applicationName = application.getName();
+                applicationJson.setExportedApplication(application);
+                return newPageRepository.findByApplicationId(applicationId, AclPermission.READ_PAGES)
+                    .collectList()
+                    .flatMap(newPageList -> {
+                        Map<String, Set<String>> mongoEscapedWidgetsNames = new HashMap<>();
+                        newPageList.forEach(newPage -> {
+                            newPageIdMap.put(newPage.getId(), newPage.getUnpublishedPage().getName());
+                            newPage.setApplicationId(applicationName);
+                            examplesOrganizationCloner.makePristine(newPage);
+
+                            if (newPage.getUnpublishedPage().getLayouts() != null) {
+                                newPage.getUnpublishedPage().getLayouts().forEach(layout ->
+                                    mongoEscapedWidgetsNames.put(layout.getId(), layout.getMongoEscapedWidgetNames())
+                                );
+                            }
+
+                            if (newPage.getPublishedPage() != null
+                                    && newPage.getPublishedPage().getLayouts() != null) {
+
+                                newPage.getPublishedPage().getLayouts().forEach(layout ->
+                                    mongoEscapedWidgetsNames.put(layout.getId(), layout.getMongoEscapedWidgetNames())
+                                );
+                            }
+                        });
+                        applicationJson.setPageList(newPageList);
+                        applicationJson.setMongoEscapedWidgets(mongoEscapedWidgetsNames);
+                        return datasourceRepository
+                            .findAllByOrganizationId(organizationId, AclPermission.READ_DATASOURCES)
+                            .collectList();
+                    })
+                    .map(datasourceList -> {
+                        datasourceList.forEach(datasource -> datasourceMap.put(datasource.getId(), datasource.getName()));
+                        applicationJson.setDatasourceList(datasourceList);
+                        return newActionRepository.findByApplicationId(applicationId);
+                    })
+                    .flatMap(Flux::collectList)
+                    .map(newActionList -> {
+                        Set<String> concernedDBNames = new HashSet<>();
+                        newActionList.forEach(newAction -> {
+                            newAction.setPluginId(pluginMap.get(newAction.getPluginId()));
+                            newAction.setOrganizationId(null);
+                            newAction.setPolicies(null);
+                            newAction.setApplicationId(applicationName);
```

Comment:
Same as a previous comment. I think we can set this to `null` like the others, right?

```suggestion
                            newAction.setApplicationId(null);
```

Reply:
Updated code pushed!